sparse/coo/umath.py

```diff
@@ -57,8 +57,7 @@ def elemwise(func, *args, **kwargs):
         elif isscalar(arg) or (isinstance(arg, np.ndarray)
                                and not arg.shape):
             # Faster and more reliable to pass ()-shaped ndarrays as scalars.
-            if isinstance(arg, np.ndarray):
-                args[i] = arg[()]
+            args[i] = np.asarray(arg)[()]
```

Comment:
Converting all arguments to Numpy-based `dtype`s so accessing their `dtype` isn't an issue.